torch/distributed/checkpoint/default_planner.py

```diff
@@ -195,6 +197,20 @@ def set_up_planner(
 
     def create_local_plan(self) -> LoadPlan:
         assert self.metadata is not None
+        if self.flatten_state_dict:
+            current_keys = set(self.state_dict.keys())
+            load_keys = set(self.metadata.state_dict_metadata.keys())
+            missing_keys = load_keys - current_keys
+            if missing_keys:
+                _version._act_like_version = "2_3"
+                old_state_dict, old_mappings = flatten_state_dict(
+                    self.original_state_dict
+                )
+                old_keys = set(old_state_dict.keys())
+                if old_keys & missing_keys:
+                    self.state_dict, self.mappings = old_state_dict, old_mappings
+                _version._act_like_version = None
```

Comment:
I do not understand why we set the 

_version._act_like_version = 2.3 and then set the same to None at the end. 
Isn't that a noop ?

Reply:
Please see the comment.